src-static/update_locations.R

```diff
@@ -114,56 +90,46 @@ region_match_df <- dplyr$tribble(
   "CAN", "North America" # Creating custom North America region
 )
 
-df_names <- df_ocha_names |>
+df_regions <- df_ocha_names |>
   dplyr$left_join(
     region_match_df,
     by = "iso3"
-  ) |>
+  )
+
+# don't use dplyr to avoid warnings for missing values in countrycode
+uses_custom_reg <- !is.na(df_regions$region_custom)
+df_regions$unhcr_region <- NA_character_
+df_regions$unhcr_region[uses_custom_reg] <- df_regions$region_custom[uses_custom_reg]
+df_regions$unhcr_region[!uses_custom_reg] <- countrycode$countrycode(
+  sourcevar = df_regions$iso3[!uses_custom_reg],
+  origin = "iso3c",
+  destination = "unhcr.region"
+)
+
+# pull together unhcr region and custom region into one final dataset
+df_locations <- df_regions |>
   dplyr$transmute(
     iso3,
-    country,
-    unhcr_region = countrycode$countrycode(
-      sourcevar = iso3,
-      origin = "iso3c",
-      destination = "unhcr.region"
-    ),
+    location,
+    location_note,
     region = dplyr$case_when(
       !is.na(region_custom) ~ region_custom,
       unhcr_region == "The Americas" ~ "Latin America and the Caribbean",
       unhcr_region %in% c("Southern Africa", "East and Horn of Africa") ~ "Southern and Eastern Africa",
       .default = unhcr_region
     )
-  ) |>
-  dplyr$filter(
-    iso3 != "ATA" # drops antarctica
-  ) |>
-  dplyr$select(
-    -unhcr_region
   )
 
-
-#######################
-#### HRP COUNTRIES ####
-#######################
-
-df_hrp <- dplyr$tibble(
-  iso3 = iso3_codes,
-  hrp_country = iso3 %in% cs$read_az_file("input/hrp_countries.parquet")$iso3
-)
-
-#######################
-#### FINAL DATASET ####
-#######################
-
-df_country_info <- purrr$reduce(
-  .x = list(df_names, df_hrp, df_centroids),
-  .f = \(x, y) dplyr$left_join(x, y, by = "iso3")
-)
-
-fname <- "input/country_info.parquet"
-cs$update_az_file(
-  df = df_country_info,
-  name = fname
-)
-
-log_info(paste0("Successfully downloaded countries info and saved to ", fname))
+if (any(is.na(dplyr$select(df_locations, -location_note)))) {
+  stop(
+    "Location date has missing values, fix before re-running.",
+    call. = FALSE
+  )
+} else {
+  fname <- "input/locations.parquet"
+  cs$update_az_file(
+    df = df_locations,
+    name = fname
```

Comment:
might be good to explicitly specify `blob` arg

Reply:
I think on this we want to rely on the default being `prod` since we only don't want to read or write from `prod` in a very rare set of circumstances. In fact, don't even want to do that long term, ideally will only work to and from `prod`.